pkg/app/utils/utils.go

```diff
@@ -125,3 +126,32 @@ func DeployApp(mgmtAppClient projv3.AppInterface, projectID string, createOrUpda
 
 	return rtn, nil
 }
+
+func DetectAppCatalogExistence(appCatalogID string, cattleTemplateVersionsClient v3.CatalogTemplateVersionInterface) error {
+	templateVersionID, templateVersionNamespace, err := common.ParseExternalID(appCatalogID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse catalog ID %q", appCatalogID)
+	}
+
+	_, err = cattleTemplateVersionsClient.GetNamespaced(templateVersionNamespace, templateVersionID, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to find catalog by ID %q", appCatalogID)
+	}
+
+	return nil
+}
+
+func DeleteApp(mgmtAppClient projv3.AppInterface, projectID, appName string) error {
+	app, err := mgmtAppClient.GetNamespaced(projectID, appName, metav1.GetOptions{})
```

Comment:
q: why are we getting the app here if it's just for delete?

Reply:
To check for `app.DeletionTimestamp == nil`

Reply:
I don't get it, why make a call to db just to check if the object is already deleted, we could just issue another delete and check for notFound error